src/main/java/frc/robot/subsystems/Drive.java

```diff
@@ -212,4 +301,51 @@ public void end(final boolean interrupted) {
         return cmd;
 
     }
+
+    public CommandBase autonomousCommand(String trajectoryName) {
+        return autonomousCommand(trajectoryName, true);
+    }
+
+    public CommandBase autonomousCommand(String trajectoryName, Boolean resetPose) {
+
+        String trajectoryJSON = "paths/" + trajectoryName + ".wpilib.json";
+        Trajectory autoTrajectory = new Trajectory();
+        try {
+            Path trajectoryPath = Filesystem.getDeployDirectory().toPath().resolve(trajectoryJSON);
+            autoTrajectory = TrajectoryUtil.fromPathweaverJson(trajectoryPath);
+        } catch (IOException ex) {
+            DriverStation.reportError("Unable to open trajectory: " + trajectoryJSON, ex.getStackTrace());
+        }
+
+        final Trajectory finalAutoTrajectory = autoTrajectory;
+
+        RamseteCommand ramseteCommand = new RamseteCommand(autoTrajectory,
+                // Gets pose
+                this::getPose,
+                // Creates our ramsete controller
+                new RamseteController(RAMSETE_B, RAMSETE_ZETA),
+                // Allows for robot control
+                new SimpleMotorFeedforward(KS_VOLTS, KV_VOLT_SPM, KA_VOLT_SSPM),
+                // Describes how the drivetrain is influenced by motor speed
+                trajectoryKinematics,
+                // Gets the speed of the wheels
+                this::getWheelSpeeds,
+                // Left Controller
+                new PIDController(DRIVE_VEL_P, 0, DRIVE_VEL_D),
+                // Right Controller
+                new PIDController(DRIVE_VEL_P, 0, DRIVE_VEL_D),
+                // Sends voltages to motors
+                this::tankDriveVolts, this);
+
+        CommandBase cmd;
+        if (resetPose) {
+            cmd = (new InstantCommand(() -> resetOdometry(finalAutoTrajectory.getInitialPose())))
```

Comment:
This could probably be made nicer, but it's not essential right now

Reply:
you mean stick it all in that second .andThen()?